source/common/http/filter_manager.h

```diff
@@ -594,6 +595,46 @@ class FilterManagerCallbacks {
   virtual const ScopeTrackedObject& scope() PURE;
 };
 
+/**
+ * This class allows the remote address to be overridden for HTTP stream info. This is used for
+ * XFF handling. This is required to avoid providing stream info with a non-const address provider.
+ */
+class OverridableRemoteSocketAddressProviderStreamInfo : public StreamInfo::StreamInfoImpl,
+                                                         Network::SocketAddressProviderGetters {
```

Comment:
Should this be publicly inherited?

If not, please explicitly specify `private` for readability.

Reply:
It should be private since we want users to access address info via the normal interface getter. I will add a comment, etc.